File: VB/GrowthWare.Framework/Model/Profiles/MClientChoicesState.vb
Imports System.Collections

Namespace Model.Profiles
    <Serializable(), CLSCompliant(True)> _
    Public Class MClientChoicesState
        Dim mClientChoices As Hashtable = New Hashtable(StringComparer.InvariantCultureIgnoreCase)
        Dim mAccountName As String
        Dim mIsDirty As Boolean = False

        Public Sub New(ByVal clientChoicesData As DataRow)
            MyBase.New()
            mAccountName = AccountName
            If Not clientChoicesData Is Nothing Then
                Try
                    Dim myTable As DataTable = clientChoicesData.Table.Copy
                    Dim Row As DataRow = myTable.Rows(0)
                    Dim i As Integer
                    For i = 0 To myTable.Columns.Count - 1
                        Dim Value As Object = Row.Item(myTable.Columns(i))
                        If myTable.Columns(i).ToString() = "ACCT" Then mAccountName = Value.ToString()
                        mClientChoices(myTable.Columns(i).ToString()) = Value.ToString()
                    Next i
                Catch
                    Throw
                End Try
            End If
        End Sub

        Public Property AccountName() As String
            Get
                Return mAccountName
            End Get
            Set(ByVal Value As String)
                mAccountName = Value
            End Set
        End Property

        Public ReadOnly Property ChoicesHashtable() As Hashtable
            Get
                Return mClientChoices
            End Get
        End Property

        Default Public Property Item(ByVal key As String) As String
            Get
                Return CType(mClientChoices(key), String)
            End Get
            Set(ByVal Value As String)
                mClientChoices(key) = Value
                mIsDirty = True
            End Set
        End Property

        Public Property IsDirty() As Boolean
            Get
                Return mIsDirty
            End Get
            Set(ByVal value As Boolean)
                mIsDirty = value
            End Set
        End Property
    End Class
End Namespace

File: CS/GrowthWare.Framework/Model/Profiles/MClientChoicesState.cs
using System;
using System.Collections;
using System.Data;

namespace GrowthWare.Framework.Model.Profiles
{
    /// <summary>
    /// Class MClientChoicesState
    /// </summary>
    [Serializable(), CLSCompliant(true)]
    public class MClientChoicesState
    {
        Hashtable m_ClientChoices = new Hashtable(StringComparer.OrdinalIgnoreCase);

        string m_AccountName;

        bool m_IsDirty = false;

        /// <summary>
        /// Initializes a new instance of the <see cref="MClientChoicesState" /> class.
        /// </summary>
        /// <param name="clientChoicesData">The client choices data.</param>
        public MClientChoicesState(DataRow clientChoicesData)
            : base()
        {
            m_AccountName = AccountName;
            if (clientChoicesData != null) 
            {
                try
                {
                    DataTable myTable = clientChoicesData.Table.Copy();
                    DataRow Row = myTable.Rows[0];
                    int i = 0;
                    for (i = 0; i <= myTable.Columns.Count - 1; i++)
                    {
                        object Value = Row[myTable.Columns[i]];
                        if (myTable.Columns[i].ToString() == "ACCT") m_AccountName = Value.ToString();
                        m_ClientChoices[myTable.Columns[i].ToString()] = Value.ToString();
                    }
                }
                catch (Exception)
                {
                    throw;
                }
            }
        }

        /// <summary>
        /// Gets or sets the name of the account.
        /// </summary>
        /// <value>The name of the account.</value>
        public string AccountName
        {
            get { return m_AccountName; }
            set { m_AccountName = value; }
        }

        /// <summary>
        /// Gets the choices hashtable.
        /// </summary>
        /// <value>The choices hashtable.</value>
        public Hashtable ChoicesHashtable
        {
            get { return m_ClientChoices; }
        }

        /// <summary>
        /// Gets or sets the <see cref="String" /> with the specified key.
        /// </summary>
        /// <param name="key">The key.</param>
        /// <returns>[ERROR: invalid expression ReturnTypeName.FullName].</returns>
        public String this[string key]
        {
            get { return (string)m_ClientChoices[key]; }
            set
            {
                m_ClientChoices[key] = value;
                m_IsDirty = true;
            }
        }

        /// <summary>
        /// Gets or sets a value indicating whether this instance is dirty.
        /// </summary>
        /// <value><c>true</c> if this instance is dirty; otherwise, <c>false</c>.</value>
        public bool IsDirty
        {
            get { return m_IsDirty; }
            set { m_IsDirty = value; }
        }
    }
}
